crates/edr_provider/src/requests/eth/transactions.rs

```diff
@@ -1,6 +1,117 @@
-use edr_eth::{serde::ZeroXPrefixedBytes, transaction::EthTransactionRequest, B256};
+use edr_eth::{
+    remote,
+    serde::ZeroXPrefixedBytes,
+    transaction::{EthTransactionRequest, SignedTransaction},
+    SpecId, B256, U256,
+};
 
-use crate::{data::ProviderData, ProviderError};
+use crate::{
+    data::{BlockMetadataForTransaction, GetTransactionResult, ProviderData},
+    ProviderError,
+};
+
+const FIRST_HARDFORK_WITH_TRANSACTION_TYPE: SpecId = SpecId::BERLIN;
+
+pub async fn handle_get_transaction_by_hash(
+    data: &ProviderData,
+    transaction_hash: B256,
+) -> Result<Option<remote::eth::Transaction>, ProviderError> {
+    data.transaction_by_hash(&transaction_hash)
+        .await?
+        .map(get_transaction_result_to_rpc_result)
+        .transpose()
+}
+
+fn get_transaction_result_to_rpc_result(
```

Comment:
We might want to introduce a separate module at some point for these RPC conversions, but seemed like an overkill for now.